File: src/main/java/gargoyle/netsquares/store/prop/PrefLongProperty.java
package gargoyle.netsquares.store.prop;

public class PrefLongProperty extends BasePrefProperty<Long> {
    public PrefLongProperty(Class<?> pkg, String key, Long defaultValue) {
        super(pkg, key, defaultValue);
    }

    @Override
    protected void _set(Long value) {
        preferences.putLong(key, value);
    }

    public Long get(Long defaultValue) {
        return preferences.getLong(key, defaultValue);
    }

    public void setValue(long value) {
        set(value);
    }
}

File: src/main/kotlin/gargoyle/netsquares/store/prop/PrefLongProperty.kt
package gargoyle.netsquares.store.prop

import kotlin.reflect.KClass

class PrefLongProperty(pkg: KClass<*>, key: String, defaultValue: Long) :
    BasePrefProperty<Long>(pkg, key, defaultValue) {
    override fun _set(value: Long) {
        preferences.putLong(key, value)
    }

    override fun get(defaultValue: Long): Long {
        return preferences.getLong(key, defaultValue)
    }

    fun setValue(value: Long) {
        set(value)
    }
}
